### ISSUE TITLE
Gradient detch for generator loss calculation

### ISSUE THREAD

**Issue by nevoliu** [taravatp/Predicting_Future_Frames_Using_Retrospective_Cycle_GAN_pytorch#1]
Hi taravatp,

Great of your work. I'm learning to use cycgan prediction videos. I have some doubts.

Why do some variables use .detach() and some variables do not use .detach() when calculating the loss of generators？
![loss](https://user-images.githubusercontent.com/119554616/210694171-b3bdf8d6-f464-4216-9a82-5fec89155383.png)
